relay-sampling/src/config.rs

```diff
@@ -512,19 +499,28 @@ mod tests {
         assert_eq!(serialized_config, expected_serialized_config)
     }
 
+    /// Checks that the sample rate stays constant if `DecayingFunction::Constant` is set.
+    #[test]
+    fn test_decay_fn_constant() {
+        let sample_rate = 0.5;
+
+        assert_eq!(
+            DecayingFunction::Constant.adjust_sample_rate(
+                sample_rate,
+                Utc::now(),
+                TimeRange::default()
+            ),
+            Some(sample_rate)
+        );
+    }
+
     /// Checks if the sample rate decays linearly if `DecayingFunction::Linear` is set.
     #[test]
-    fn test_sample_rate_with_linear_decay() {
-        let rule = SamplingRule {
-            condition: RuleCondition::all(),
-            sampling_value: SamplingValue::SampleRate { value: 1.0 },
-            ty: RuleType::Trace,
-            id: RuleId(0),
-            time_range: TimeRange {
-                start: Some(Utc.with_ymd_and_hms(1970, 10, 10, 0, 0, 0).unwrap()),
-                end: Some(Utc.with_ymd_and_hms(1970, 10, 12, 0, 0, 0).unwrap()),
-            },
-            decaying_fn: DecayingFunction::Linear { decayed_value: 0.5 },
+    fn test_decay_fn_linear() {
```

Comment:
Now that decaying function logic is within the decayingfunction type, we can unit test more directly where the logic resides.